core/src/test/java/io/opencensus/stats/MeasureMapTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2016, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.stats;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import io.opencensus.stats.Measure.DoubleMeasure;
+import java.util.ArrayList;
+import java.util.Iterator;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Tests for {@link MeasureMap}
+ */
+@RunWith(JUnit4.class)
+public class MeasureMapTest {
+
+  @Test
+  public void testOf1() {
+    ImmutableList<Measurement> expected = ImmutableList.of(
+        (Measurement) Measurement.DoubleMeasurement.create(M1, 44.4));
```

Comment:
Is there any way that we can avoid the cast when adding Measurements to a collection?